src/components/Popup/Popup.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+import CSSProperties = React.CSSProperties
+
+import { childrenExist, customPropTypes, isBrowser } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import { ItemShorthand, Extendable } from '../../../types/utils'
+import Portal from '../Portal'
+import PopupContent from './PopupContent'
+
+type PopupPosition =
+  | 'top start'
+  | 'top end'
+  | 'bottom start'
+  | 'bottom end'
+  | 'start center'
+  | 'end center'
+  | 'top center'
+  | 'bottom center'
+
+const POSITIONS: PopupPosition[] = [
+  'top start',
+  'top end',
+  'bottom start',
+  'bottom end',
+  'start center',
+  'end center',
+  'top center',
+  'bottom center',
+]
+
+export interface IPopupProps {
+  as?: any
+  basic?: boolean
+  className?: string
+  content?: ItemShorthand | ItemShorthand[]
+  position?: PopupPosition
+  style?: CSSProperties
+  trigger: JSX.Element
+}
+
+export interface IPopupState {
+  style?: CSSProperties
+}
+
+/**
+ * A Popup displays additional information on top of a page.
+ * @accessibility This is example usage of the accessibility tag.
+ * This should be replaced with the actual description after the PR is merged
+ */
+export default class Popup extends React.Component<Extendable<IPopupProps>, IPopupState> {
+  private popupOffset = 8
```

Comment:
Is there any plan to make ```offset``` a prop? So a user can set it as he wants

Reply:
yup, we will do that in next iterations of popup